tests/unit/test_errorfactory.py

```diff
@@ -0,0 +1,30 @@
+import sys
+
+from nose.tools import assert_equals, assert_true
+
+from botocore.exceptions import ClientError
+from botocore.errorfactory import ServiceErrorFactory
+from botocore.model import ServiceModel
+
+
+def test_errorfactory():
+    model = {
+        'operations': {
+            'OperationName': {
+                'name': 'OperationName',
+                'errors': [{'shape': 'NoSuchResourceException'}],
+            }
+        },
+        'shapes': {
+            'NoSuchResourceException': {
+                'type': 'structure',
+                'members': {}
+            }
+        }
+    }
+    factory = ServiceErrorFactory(ServiceModel(model))
+    members = dir(factory)
+    if sys.version_info >= (2, 7):
```

Comment:
Awesome thanks for the deep dive on that! Yeah I am not a big fan of accommodating python 2.6 as well, but it is a backwards compatibility thing; since botocore started off officially supporting python2.6, we need to continue supporting it until we make a stance to stop supporting 2.6, which in itself will take a decent amount of time.